Aims to fix conda envs issue.

pmb59 opened the pull request.

pmb59 committed:
Aims to fix conda envs issue.

pcm32 commented:
I made this comment elsewhere, but for you convenience:

you could get rid of all the python mumbo-jumbo and just have something like:

```
rule env_1:
   env: envs/env1.yaml
   shell:
     """
     echo "I'm easy 1"
     """

rule env_2:
   env: envs/env2.yaml...
```

and then run that as dry run with the conda creation flag.

pcm32 commented:
...mmm... or even better, use the python API from Snakemake and iterate over all existing yamls in envs and send those to build (if they don't exist ideally)... then there is no tracking needed of new YAMLs added. Shortcoming of this could be re-generating (and failing) of existing working conda envs (in time, this tends to happen, that is why in another topics we tend to use singularity).

pmb59 committed:
Iterates over all existing yamls in /envs and send those to build in a --dry-run

pcm32 commented:
I think that this PR is obsolete compared to what you have currently, right @pmb59 ?

pmb59 committed:
These fixes are now up-to-date re creation of conda environments in a dry run.

pmb59 committed:
Adds  --latency-wait parameter to the snakemake call to minimise missing file errors probably be due to filesystem latency.

pmb59 committed:
Removes unnecessary code. If only one file has been created (RNA-seq case), then it is already there.

In Mac this has no effect as the output of doing `mv` with destination equal to source is 0, but in Linux is 1 therefore raising an error.

pmb59 committed:
Resolves conflict with develop

pcm32 commented on Snakefile:
```diff
@@ -201,8 +201,8 @@ rule percentile_ranks:
-        else
-            mv ${{percentile_ranks[0]}} {output.percentile_ranks_merged}
+            #else
+            #mv ${{percentile_ranks[0]}} {output.percentile_ranks_merged}
```
Lets just remove this. Although strictly speaking, I would suggest a separate PR for this, then it is easier to revert if needed (and safer to remove the comments).

pcm32 replied:
But given the small change, lets just leave it here.

pmb59 replied:
Done

pcm32 commented on run_sorting_hat_test_data.sh:
```diff
@@ -10,22 +10,22 @@ SN_CONDA_PREFIX=${SN_CONDA_PREFIX:-$( pwd )/conda_installs}
+#create conda envs only
+for yaml_file in $(ls $( pwd )/envs); do
+  echo $yaml_file
+  snakemake --dry-run --use-conda --conda-create-envs-only $CONDA_PREFIX_LINE  \
```
I think that --dry-run is redundant when using --conda-create-envs-only, on the successful run (see slack 02/09/2021 ~2:44) I didn't use --dry-run.

pmb59 replied:
done

pcm32 commented on Snakefile-sorting-hat:
```diff
@@ -165,12 +165,15 @@ rule produce_recalculations_call:
             metadata_summary=../{input.metadata} \
             bioentities_properties={params.bioentities_properties} \
             delete_previous_output={params.delete_previous_output} \
+            skip_steps_file={params.skip_steps_file} \
```
Could you remind me what was this about? was it to skip steps for datasets where we don't have raw sequences?

pmb59 replied:
This is regarding a non-public YAML file with a reduced list of experiments that specifically can skip certain steps. This is necessary in absence of certain files.

pcm32 reviewed (approved):
LGTM, left a couple of minor comments, feel free to merge after addressing them.

pmb59 committed:
Addresses PR review comments
